### PR TITLE
Added saving traces to file.

### PR DESCRIPTION
**Pull request by dbaghdasarya**

### CODE REVIEW

File: src/main/java/com/wavefront/Span.java [hunk 1]
```diff
@@ -116,4 +118,39 @@ public boolean removeParent(Span parent) {
...
+
+    sb.append(" ").append(startMillis);
+    sb.append(" ").append(durationMillis);
+    sb.append("\n");
+
+    if (children != null) {
+      children.forEach(child -> sb.append(child.toString()));
+    }
+    return sb.toString();
+  }
```
Comment (sirak92) [inline src/main/java/com/wavefront/Span.java:119]: Please use tracingSpanToLineData function from com.wavefront.sdk.common.Utils.

File: src/main/java/com/wavefront/SpanSender.java [hunk 2]
```diff
@@ -2,25 +2,58 @@
...
+import com.google.common.cache.LoadingCache;
+
 /**
  * TODO
  *
  * @author Davit Baghdasaryan (dbagdasarya@vmware.com)
  */
 public class SpanSender {
   private static final Logger logger = Logger.getLogger(SpanSender.class.getCanonicalName());
+  private static LoadingCache<String, FileWriter> fileWriterLoadingCache = CacheBuilder.newBuilder()
```
Comment (sirak92) [inline src/main/java/com/wavefront/SpanSender.java]: I don't see the benefit of using LoadingCache here, spanQueue is written to the file once, so there is no need to keep FileWriter in the cache.

File: src/main/java/com/wavefront/SpanSender.java [hunk 3]
```diff
@@ -2,25 +2,58 @@
...
+      .build(new CacheLoader<>() {
+        @Override
+        public FileWriter load(String fileName) throws Exception {
+          return new FileWriter(new File(fileName));
+        }
+      });
+
+  private FileWriter fileWriter;
```
Comment (sirak92) [inline src/main/java/com/wavefront/SpanSender.java]: again filewriter can be local

File: src/main/java/com/wavefront/SpanSender.java [hunk 4]
```diff
@@ -51,4 +84,15 @@ public void startSending(double rate, SpanQueue spanQueue) throws IOException, I
 
     logger.info("Sending complete!");
   }
+
+  private void saveToFile(SpanQueue spanQueue) throws ExecutionException, IOException {
+    fileWriter = fileWriterLoadingCache.get(outputFile);
+    int sent = 0;
```
Comment (sirak92) [inline src/main/java/com/wavefront/SpanSender.java]: spansSize = spanQueue.size();
no need to increment to know the size.

File: src/main/java/com/wavefront/SpanSender.java [hunk 5]
```diff
@@ -51,4 +84,15 @@ public void startSending(double rate, SpanQueue spanQueue) throws IOException, I
...
+    Span tempSpan;
+    while ((tempSpan = spanQueue.pollFirst()) != null) {
+      fileWriter.write(tempSpan.toString());
+      sent++;
+    }
+    logger.info(sent + " spans saved to file!");
```
Comment (sirak92) [inline src/main/java/com/wavefront/SpanSender.java]: File name information in log will be useful, isn't it?

File: src/main/java/com/wavefront/WavefrontTraceLoader.java [hunk 6]
```diff
@@ -29,16 +31,20 @@ void setupSenders() throws IOException {
     }
     // TODO do we need additional checks here??
     WavefrontSender wavefrontSender;
```
Comment (sirak92) [inline src/main/java/com/wavefront/WavefrontTraceLoader.java]: wafefrontSender definition can go to else body.

File: src/main/java/com/wavefront/Span.java (hunk 1 above)
Comment (dbaghdasarya) [reply src/main/java/com/wavefront/Span.java:119]: Done. Thank you for the hint!

File: src/main/java/com/wavefront/SpanSender.java (hunk 2 above)
Comment (dbaghdasarya) [reply src/main/java/com/wavefront/SpanSender.java]: Agree

File: src/main/java/com/wavefront/WavefrontTraceLoader.java (hunk 6 above)
Comment (dbaghdasarya) [reply src/main/java/com/wavefront/WavefrontTraceLoader.java]: Done

**Pull request merged**